mlflow/utils/virtualenv.py

```diff
@@ -237,15 +238,32 @@ def _create_virtualenv(
 
     _logger.info("Installing dependencies")
     for deps in filter(None, [python_env.build_dependencies, python_env.dependencies]):
-        # Create a temporary requirements file in the model directory to resolve the references
-        # in it correctly.
-        tmp_req_file = f"requirements.{uuid.uuid4().hex}.txt"
-        local_model_path.joinpath(tmp_req_file).write_text("\n".join(deps))
-        try:
+        with TempDir() as t:
+            # Create a temporary requirements file in the model directory to resolve the references
+            # in it correctly. To do this, we must first symlink or copy the model directory's
+            # contents to a temporary location for compatibility with deployment tools that store
+            # models in a read-only mount
+            model_dir = t.path("model")
+            try:
+                os.makedirs(model_dir)
+                for model_item in os.listdir(local_model_path):
+                    os.symlink(
```

Comment:
Do we have to specify `target_is_directory` arg for Windows since symbolic links are handled autonomously only in developer mode (I think there are 2 distinct implementations based on whether the target (src) is a file or a directory, but I may be wrong here...)?

Reply:
Looks like we don't based on the following: 

```
On Windows, a symlink represents either a file or a directory, and does not morph to the target dynamically. If the target is present, the type of the symlink will be created to match. 
```